### DISCUSSION TITLE
How to remove image from fltk::dialog::message?

### DISCUSSION THREAD

**Discussion by abau** [fltk-rs/fltk-rs#1167] [Q&A] [answered]
How to remove image from fltk::dialog::message? None of the following works for me:
```
message_icon().hide();

unsafe { WidgetBase::delete(message_icon().into_widget::<Frame>()) };

message_icon().set_image::<SvgImage>(None);
```

**Comment by MoAlyousef** [comment, marked as answer]
It's not an image that's shown by default, just a Frame with a special label.
You can try:
```rust
        let mut i = dialog::message_icon();
        i.set_frame(FrameType::NoBox);
        i.set_label("");
        message_default("Hello");
```

**Reply by abau** [reply]
> It's not an image that's shown by default, just a Frame with a special label.

Ah, now I get it. That makes sense. 

With your code, there still is an empty space where previously the label text/image was rendered. Is there a way to remove this space as well (`dialog::message_icon().set_size(0, 0)` does not work)?

Thank you!

**Reply by MoAlyousef** [reply]
Unfortunately changing the icon’s size isn’t supported (from the FLTK docs):
https://github.com/fltk/fltk/blob/d9a6ec88e47b9c3c7993018f4f2be13d55b8fca2/src/fl_ask.cxx#L274

If it’s a simple dialog message box, I would suggest creating your own, it’s quite simple:
```rust
use fltk::{enums::*, prelude::*, *};

fn my_message_default(label: &str) {
    let mut win = window::Window::default()
        .with_size(300, 100)
        .with_label("Dialog");
    frame::Frame::default()
        .with_size(250, 90)
        .center_of_parent()
        .with_label(label)
        .with_align(Align::Left | Align::Inside);
    win.end();
    win.make_modal(true);
    win.show();
    while win.shown() {
        app::wait();
    }
}
```

**Reply by abau** [reply]
> Unfortunately changing the icon’s size isn’t supported

I see.

> I would suggest creating your own, it’s quite simple:

Ok, will do. Thanks again!
